libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -791,10 +797,12 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
     NSURL *url = navigationAction.request.URL;
     NSString *requestUrl = [url absoluteString];
     if ([self isRedirectURL:requestUrl]) {
-        if ([[SalesforceSDKManager sharedManager] useWebServerAuthentication]) {
-            [self handleWebServerResponse:url];
+        // Determine if presence of override parameters requiring the user agent flow.
+        BOOL overrideWithUserAgentFlow = self.overrideWithfrontDoorBridgeUrl && !self.overrideWithCodeVerifier;
+        if ( [[SalesforceSDKManager sharedManager] useWebServerAuthentication] && !overrideWithUserAgentFlow) {
```

Comment:
@bbirman, @brandonpage and @wmathurin - This works, but I wonder if it's the best way to "override" to the flow needed by the QR code Vs. the app's preference.  I tested both QR code types plus logging in as usual via the web view.  It gives me the control flow I expected, but I would love more eyes on this spot.

If it helps to paraphrase the logic, the two "override" parameters are only present when calling through the auth helper with them from a QR code log in.  A front door URL by itself needs user agent flow and with a code verifier requires web server flow.

Perhaps it's coincidental, but I noticed the user agent flow uses a fragment string `#` and the web server flow uses a query string `?`.  I tried it out, just switching on that, and it actually works perfectly since these two paths are hard coded to those URL formats.  Would it be simpler to switch on the URL format?

Reply:
I don't know if we should count on the URL format staying the same. I would rather having it based on something we truly control (like the presence or absence of the code verifier in the QR url). 

Alternatively, we could have another parameter encoded in the QR url to indicate explicitly the flow to use: useUserAgentFlow = true | false.

Or like you said, we could just fall back on the the app configuration itself in which case a QR url without code verifier in an app configured to use web server flow, would end up using web server flow without pkce.

What am I saying? I think we can start with the approach you implemented above (driven by the QR url but without a dedicated param).